src/test/java/com/lezurex/overcomplicated/MaybeTest.java

```diff
@@ -0,0 +1,18 @@
+package com.lezurex.overcomplicated;
+
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * @author Z-100
+ * This test may or may not fail
+ */
+class MaybeTest {
+
+	@Test
+	void maybe() {
+		assertTrue(Maybe.maybe());
+		assertFalse(Maybe.maybe());
+	}
```

Comment:
Random values cannot be tested. Either remove this test or mock the random values (somehow).